packages/harmony/src/components/input/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,213 @@
+import { forwardRef, type ComponentPropsWithoutRef } from 'react'
+
+import cn from 'classnames'
+
+import layoutStyles from 'components/layout/layout.module.css'
+import { IconComponent, Text } from 'components/typography'
+
+import { HelperText } from './HelperText'
+import styles from './TextInput.module.css'
+import { useFocusState } from './useFocusState'
+
+export enum TextInputSize {
+  SMALL = 'small',
+  DEFAULT = 'default'
+}
+
+export type TextInputProps = Omit<
+  ComponentPropsWithoutRef<'input'>,
+  // Omitting required purely for storybook docs
+  'size' | 'required'
+> & {
+  /**
+   * Input sizes. NOTE: small inputs will not show the label
+   * @default default
+   */
+  size?: TextInputSize
+  /**
+   * Show a "x/Max" text displaying number of characters allowed
+   */
+  showMaxLength?: boolean
+  /**
+   * Toggle warning state
+   */
+  warning?: boolean
+  /**
+   * Toggle error state
+   */
+  error?: boolean
+  /**
+   * Hides the label. If the label is hidden the placeholder will show instead.
+   * @default false
+   */
+  hideLabel?: boolean
+  /**
+   * ClassName on the div wrapping the whole input container (doesn't include assistive text)
+   */
+  inputRootClassName?: string
+  /**
+   * Label Text
+   */
+  label?: string
+  /**
+   * Helper text that shows up below the input
+   */
+  helperText?: string
+  /**
+   * Floating text on the lefthand side of the input.
+   */
+  startAdornmentText?: string
+  /**
+   * Floating text on the righthand side of the input
+   */
+  endAdornmentText?: string
+  /**
+   * Floating icon on the lefthand side of the input. Note: will float to the left of the label & content
+   */
+  startIcon?: IconComponent
+  /**
+   * Floating text on the righthand side of the input
+   */
+  endIcon?: IconComponent
+  /**
+   * Required or not. Will add an * to the label if required
+   */
+  required?: boolean
+}
+
+export const TextInput = forwardRef<HTMLInputElement, TextInputProps>(
+  (props: TextInputProps, ref) => {
```

Comment:
techincally dont need to put the type on props here, but maybe you do to make harmony work? also can you double check in the react dev-tools if text-input gets named collectly with forwardRef? what weve don in the past is do `const A = forwardRef(function A() {...})`

Reply:
Checked this and the name is still showing fine in devtools (at least in storybook)

Reply:
Awesome, thanks. Just realized the place to check is in web after the build. I think it might get lost at that point?